### PR TITLE
PORTALS-4134

### PR DESCRIPTION
**Pull request by jay-hodgson**
This PR adds a SearchQueryWrapper / SearchQueryWrapperPlotNav stack that drives the existing component library from the SearchQueryServicesApi rather than the standard table query API, and wires up the b2ai.standards portal as a consumer

### CODE REVIEW

**Comment by jay-hodgson** [comment]
Helpful summary of changes from Claude Sonnet:

## PR Summary: SearchQueryServicesApi Integration Layer (PORTALS-4134)

This PR introduces a new adapter layer that allows portal pages to use Synapse's **SearchQueryServicesApi** (full-text search) as a drop-in replacement for the standard Table Query API, while reusing all existing UI components (`QueryWrapperPlotNav`, `SynapseTable`, facet filters, etc.) without modification.

### Motivation

The B2AI Standards Registry portal needs to query a `SearchIndex` entity via the dedicated Search API rather than the standard `QueryBundleRequest` table API. This PR builds the translation/adapter layer to make that possible, and wires it up in the portal framework so any portal tab can opt in.

---

### New Files

| File | Description |
|---|---|
| `packages/synapse-react-client/src/components/SearchQueryWrapper/SearchQueryUseQueryOptions.ts` | Core adapter layer. Converts a `QueryBundleRequest` (used internally for UI state) into a `SearchIndexQuery` for the Search API. Implements `toSearchIndexQuery()` (maps facet filters, range filters, sort, FTS text → Search API equivalents) and `useSearchQueryUseQueryOptions()` (builds TanStack Query `useQueryOptions` objects mirroring the `TableQueryUseQueryOptions` shape). |
| `packages/synapse-react-client/src/components/SearchQueryWrapper/SearchQueryWrapper.tsx` | React context provider that wraps children with a `QueryContext` powered by the Search API. Resolves the underlying Table/View entity from the `SearchIndex`'s `definingSQL` to obtain column models. Exposes `getRowDataQueryOptionsForPage` on the context so prefetch works correctly without calling the table API. |
| `packages/synapse-react-client/src/components/SearchQueryWrapper/SearchQueryUseQueryOptions.test.ts` | Unit tests for the `toSearchIndexQuery` conversion function and `useSearchQueryUseQueryOptions` hook, covering facet filters, range filters, sort, FTS text, pagination, and edge cases. |
| `packages/synapse-react-client/src/components/SearchQueryWrapperPlotNav/SearchQueryWrapperPlotNav.tsx` | High-level component combining `SearchQueryWrapper` + `QueryWrapperPlotNavContents`. Accepts `SearchQueryWrapperPlotNavProps` (notably `searchIndexId` and `autocompleteFieldName`) and includes an `isSearchQueryWrapperPlotNavProps` type guard used by the portal framework dispatcher. |
| `packages/synapse-react-client/src/components/SearchQueryWrapperPlotNav/SearchQueryWrapperPlotNav.stories.tsx` | Storybook story for `SearchQueryWrapperPlotNav` using mock Search API handlers. |
| `packages/synapse-react-client/src/mocks/mockSearchQueryData.ts` | Mock data for tests/stories: a `SearchIndex` entity (`syn60001`), its underlying table (`syn60000`), a sample `SearchIndexQuery`, and a `QueryResultBundle` with realistic rows, column models, and facets. |
| `packages/synapse-react-client/src/mocks/msw/handlers/searchQueryHandlers.ts` | MSW request handlers that intercept async start/get calls to `/repo/v1/search/query/async/{start,get}` and return the mock `QueryResultBundle`. |

---

### Modified Files — Key Changes

**`synapse-react-client` — Core library**

- **`FullTextSearch.tsx`** — Added MUI `Autocomplete` support. When `ftsConfig.getSuggestions` is provided, the text input renders as an autocomplete with debounced suggestion fetching (300 ms). Selecting a suggestion immediately fires the search. The `HelpPopover` is now conditionally rendered (only when `searchHelpURL` is set).
- **`QueryContext.tsx`** — Added optional `getRowDataQueryOptionsForPage` factory to the context type, allowing `SearchQueryWrapper` to inject its own TanStack Query options so downstream prefetch logic doesn't call the wrong API.
- **`QueryWrapperPlotNav.tsx`** — Exported `QueryWrapperPlotNavContents` and its props type so `SearchQueryWrapperPlotNav` can reuse the full UI without re-implementing it. Made `remount` prop optional (defaults to no-op).
- **`UseRowSet.ts` / `usePrefetchTableData.ts`** — Prefetch logic now checks `getRowDataQueryOptionsForPage` from context before falling back to the standard table query path, preventing spurious calls to the table API with a synthetic entity ID.
- **`SynapseTableRenderers.tsx`** — Column type resolution now prefers `columnModels` (from entity bundle) over `selectColumns` when both are available, correcting type mismatches that arise when `SearchQueryResults` fabricates `STRING` types for all columns.
- **`SearchV2.tsx`** — Added `getSuggestions` to `FTSConfig` type.
- **`FacetFilterControls.tsx`** — Defensive null-checks (`columnModels ?? []`) for the case where column models have not yet loaded.
- **`QueryVisualizationWrapper.tsx`** — Passes through `hideSearchBarControl` prop.
- **`SynapseClient.ts` (SRC)** — Added `getSearchQueryAsyncJobResults()`: posts to `/search/query/async/start`, polls `/search/query/async/get/{token}`, and returns the typed async job status.
- **`KeyFactory.ts`** — Added `getSearchQueryResultWithAsyncStatusQueryKey()` to generate stable TanStack Query cache keys for Search API requests.
- **`APIConstants.ts`** — Added `SEARCH_QUERY_ASYNC_START` and `SEARCH_QUERY_ASYNC_GET` URL constants.
- **`components/index.ts`** — Re-exports `SearchQueryWrapper` and `SearchQueryWrapperPlotNav`.
- **`mocks/entity/index.ts`** / **`mocks/msw/handlers.ts`** — Registered mock search index entity and search query MSW handlers.

**`synapse-client` package**

- **`SynapseClient.ts`** — Instantiated and exposed `searchManagementServicesClient` (`SearchManagementServicesApi`) alongside the existing `searchServicesClient`.

**`synapse-types` package**

- **`index.ts`** — Added a comment noting that `SearchQueryServices` types now live in `@sage-bionetworks/synapse-client`.

**`synapse-portal-framework`**

- **`SearchParamAwareQueryWrapperPlotNav.tsx`** — Refactored into a dispatcher + two sub-components. Now accepts either `standaloneQueryWrapperProps` (original path) or `searchQueryWrapperPlotNavProps` (new Search API path). Each sub-component handles the visible/hidden (count-only) rendering for its respective API. Added `onQueryResultBundleChange` propagation to all paths.
- **`PortalSearchPage.tsx`** — `configs` array now accepts `SearchQueryWrapperPlotNavProps` in addition to `QueryWrapperPlotNavProps`. Uses the `isSearchQueryWrapperPlotNavProps` type guard to route to the correct `SearchParamAwareQueryWrapperPlotNav` variant.
- **`ResourceSearchPage.tsx`** — Minor: passes `onQueryResultBundleChange` through.

**`apps/portals/b2ai.standards`**

- **`standards.ts`** — Added `standardsSearchQueryWrapperPlotNavProps` wired to the production Search Index (`syn74909093`) with autocomplete on `name`, topic bar chart, and column link config.
- **`searchConfig.tsx`** — Switched the portal search page to use `standardsSearchQueryWrapperPlotNavProps` (Search API) instead of the previous `standardsQueryWrapperPlotNavProps` (Table API).

File: packages/synapse-react-client/src/components/FullTextSearch/FullTextSearch.tsx [hunk 1]
```diff
@@ -22,11 +29,43 @@ export function FullTextSearch({
...
   const { showSearchBar } = useQueryVisualizationContext()
   const [searchText, setSearchText] = useState('')
+  const [suggestions, setSuggestions] = useState<string[]>([])
   const searchInputRef = useRef<HTMLInputElement>(null)
 
   const { data } = useGetQueryMetadata()
   const columnModels = data?.columnModels
 
+  // Fetch autocomplete suggestions whenever searchText changes (debounced)
+  useEffect(() => {
```
Comment (hallieswan) [inline packages/synapse-react-client/src/components/FullTextSearch/FullTextSearch.tsx]: Could consider using `useDebouncedEffect` here

File: packages/synapse-react-client/src/components/FullTextSearch/FullTextSearch.tsx [hunk 2]
```diff
@@ -55,56 +85,117 @@ export function FullTextSearch({
...
+                            ref?: React.Ref<HTMLInputElement>
+                          }
+                        if (typeof ref === 'function') {
+                          ref(node)
+                        } else if (ref && 'current' in ref) {
+                          ;(
+                            ref as React.MutableRefObject<HTMLInputElement | null>
+                          ).current = node
+                        }
+                      },
```
Comment (hallieswan) [inline packages/synapse-react-client/src/components/FullTextSearch/FullTextSearch.tsx]: Could replace with the TextField's built-in `inputRef`, e.g. `inputRef={searchInputRef}`

File: packages/synapse-react-client/src/components/SearchQueryWrapper/SearchQueryUseQueryOptions.ts [hunk 3]
```diff
@@ -0,0 +1,459 @@
...
+    responseParts: new Set(['HITS', 'SELECT_COLUMNS'] as const),
+  }
+  const metadataQuery: SearchIndexQuery = {
+    ...toSearchIndexQuery(queryBundleRequest, searchIndexId, columnModels),
+    // Request all opt-in parts needed for the metadata response:
+    // TOTAL_HITS → queryCount (drives tab count / spinner resolution)
+    // SELECT_COLUMNS → column headers for the table
+    // FACETS → facet aggregations for FacetFilterControls / plots
+    responseParts: new Set(['TOTAL_HITS', 'SELECT_COLUMNS', 'FACETS'] as const),
+  }
```
Comment (hallieswan) [inline packages/synapse-react-client/src/components/SearchQueryWrapper/SearchQueryUseQueryOptions.ts:275]: Could extract the `toSearchIndexQuery(...)` call to a base query variable that's shared by both `rowDataQuery` and `metadataQuery`

File: packages/synapse-react-client/src/components/SearchQueryWrapper/SearchQueryWrapper.tsx [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+      columnModels,
+    ],
+  )
+
+  // TODO: useDeepCompareMemoize cannot meaningfully compare functions — it falls back to
+  // reference equality for them. The callbacks below (getCurrentQueryRequest, removeSelectedFacet,
+  // getRowDataQueryOptionsForPage, etc.) behave identically to a shallow useMemo dep.
+  // The correct fix is to deep-compare only serializable data fields and use a plain useMemo for
+  // the full context object. QueryWrapper.tsx has the same issue; fix both together in a follow-up.
+  const context: QueryContextType = useDeepCompareMemoize({
```
Comment (hallieswan) [inline packages/synapse-react-client/src/components/SearchQueryWrapper/SearchQueryWrapper.tsx:231]: Would it be worth tracking this work in a ticket?

**Review by hallieswan — APPROVED** [review]
This looks great! Just a couple suggestions. 

Also, the conversions between the search query and the query result bundle are well tested, but I was also wondering if it would be useful to add a test for `SearchQueryWrapper` or `SearchQueryWrapperPlotNav` to ensure that everything is correctly wired up end-to-end? We could probably reuse the mocks setup for the stories

**Review by hallieswan — APPROVED** [review]
